### PR TITLE
DO-175 company related historical founders view

### PR DESCRIPTION
**Pull request by Iva-khar**

### CODE REVIEW

**Review by roman-tiukh — CHANGES_REQUESTED** [review]
1. order by version date `desc`
2. add filter by date (FilterSet + DateFromToRangeFilter)

File: business_register/views/company_views.py [hunk 1]
```diff
@@ -69,9 +70,22 @@ class HistoricalCompanyDetailView(RegisterViewMixin,
 class HistoricalFounderView(RegisterViewMixin,
                             CachedViewMixin,
                             viewsets.ReadOnlyModelViewSet):
+    lookup_field = 'company'
     queryset = HistoricalFounder.objects.all()
     serializer_class = HistoricalFounderSerializer
 
+    def retrieve(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+        queryset = queryset.filter(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (Iva-khar) [reply business_register/views/company_views.py]: 1 & 2 - done. For the right naming I've change pk to company_id and that necessitated to override lookup_field in next comment

File: business_register/views/company_views.py [hunk 2]
```diff
@@ -69,9 +70,22 @@ class HistoricalCompanyDetailView(RegisterViewMixin,
 class HistoricalFounderView(RegisterViewMixin,
                             CachedViewMixin,
                             viewsets.ReadOnlyModelViewSet):
+    lookup_field = 'company'
```
[the remark this answers is not in the document]
Comment (Iva-khar) [reply business_register/views/company_views.py]: changed to `lookup_field = 'company_id'`

File: business_register/views/company_views.py [hunk 3]
```diff
@@ -69,8 +70,25 @@ class HistoricalCompanyDetailView(RegisterViewMixin,
 class HistoricalFounderView(RegisterViewMixin,
                             CachedViewMixin,
                             viewsets.ReadOnlyModelViewSet):
+    lookup_field = 'company_id'
     queryset = HistoricalFounder.objects.all()
     serializer_class = HistoricalFounderSerializer
+    filter_backends = [DjangoFilterBackend, OrderingFilter]
+    ordering_fields = ['history_date']
+    ordering = ['-history_date']
```
Comment (roman-tiukh) [inline business_register/views/company_views.py:78]: why you use OrderingFilter here?
Just add order to queryset

Comment (roman-tiukh) [reply business_register/views/company_views.py:78]: if you need to have ordering filter add it to FilterSet

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/views/company_views.py (hunk 3 above)
Comment (Iva-khar) [reply business_register/views/company_views.py:78]: +

Comment (Iva-khar) [reply business_register/views/company_views.py]: done

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**